app/assets/javascripts/hubstats/application.js

```diff
@@ -14,45 +14,83 @@
 //= require jquery_ujs
 //= require select2
 //= require hubstats/bootstrap
+//= require bootstrap-datepicker
 //= require_tree .
 
 $(document).ready( function() {
+  defaultDateRange()
   setDateRange()
 });
 
+function defaultDateRange() {
+  $('.input-daterange').datepicker({
+    "todayHighlight": true,
+    "endDate": "Today",
+    "todayBtn": true
+  });
+};
+
 function setDateRange() {
-  var index = readCookie("hubstats_index") || 2;
-  var timer = document.getElementById("time-select");
+  if ((readCookie("hubstats_index") === "null~~null") || (readCookie("hubstats_index") === null)) {
+    var index = getDefaultDateRange();
+  } else {
+    var index = readCookie("hubstats_index")
+  }
 
-  timer.selectedIndex = index;
+  var timer = document.getElementById("submitDateRange");
+  dates = index.split("~~")
```

Comment:
`dates` needs to be declared before you use it otherwise it ends up getting defined as global variable which is bad practice and creates the potential for a conflict.